### PR TITLE
test: Replace demo toggles with buttons in nav

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** N/A

## Summary

test: Replace demo toggles with buttons in nav #2650

- Gets rid of floating toggle panel
- Replace toggles template with nav
- Use template file for common styles and toggling page properties
- Add HTML5 semantic elements to pass aXe

image

### CODE REVIEW

File: src/demos/_assets/nav.html
Comment (driskull) [inline src/demos/_assets/nav.html:1]: We could call this file `demo-template.html` and `demo-template.ts` if that makes more sense.

**Comment by driskull** [comment]
If this is approved, we should update each demo to make sure it's passing aXe and will therefore ensure more accurate accessibility testing results. cc @jcfranco @julio8a

**Review by julio8a — CHANGES_REQUESTED** [review]
If we're going to move forward with something like this @bstifle had a nice design for the toggles and the header section for all demo pages:

image


image


image

However, @caripizza mentioned that having the `dir` toggle doesn't work as well. I don't understand the technical detail, but @caripizza, you can chime in here.

**Comment by driskull** [comment]
For the top bar, we only probably need to toggle the dark theme and RTL. Toggling everything `loading` or `disabled` doesn't make sense.

**Comment by asangma** [comment]
This is great! I'd be happy to lose the floating annoyance.
I agree that we probz don't need an app-level toggle for disabled and loading. That seems like something the test app can add per component.

re: toggles
It might be nice to have the buttons be switches labelled "Dark theme" and "RTL". And have them be off by default.

**Comment by driskull** [comment]
I'll update the HTML to use the `calcite-switch`'s

**Comment by asangma** [comment]
> mentioned that having the dir toggle doesn't work as well

@driskull Will it work to set the `dir` on `main` so that the toggles don't move?

**Comment by asangma** [comment]
@julio8a Is the full design needed for this update? If so, do you know the components for that design?

**Comment by julio8a** [comment]
No, full design not needed. Manahari has been working with Bryan on the component design for all the demo pages.

**Comment by driskull** [comment]
Updated header with a rough design. Let me know if thats a good enough starting point @asangma

**Comment by bstifle** [comment]
> For the top bar, we only probably need to toggle the dark theme and RTL. Toggling everything `loading` or `disabled` doesn't make sense.

agreed on this.

I think leaving them switches is best. more clear ux. 


Also, that header section should be sticky to the top. currently you have to scroll all the way to the top of the page to go home. (would also keep the switches readily accessible)

**Comment by driskull** [comment]
If someone could take over the styling part that would be awesome. 🌈

**Comment by julio8a** [comment]
@ManahariDahal can take over this one and work with @bstifle.

**Comment by asangma** [comment]
@julio8a I can take on the styling of the demo header.

**Comment by julio8a** [comment]
Okay, please work with @bstifle if needed.

**Comment by asangma** [comment]
@driskull Aside from setting `display: none` on the header, is there an easy way we can turn off the demo header when need be?

The shell demos (full page demos) suffer from having the demo header in there.

**Comment by driskull** [comment]
@asangma what did you have in mind? Do you want a url param, escape key to close, close button on the header?

**Comment by driskull** [comment]
A close button on the header is probably the best solution.

**Comment by bstifle** [comment]
Looking super good pals. love it! 💯

**Comment by asangma** [comment]
@driskull Hmmm...a URL param might be nice so that one doesn't have to hit a close button every time. Could also just be a js var.

**Comment by driskull** [comment]
Can we add a close button to the UI and I can add url param support after merge?

**Comment by asangma** [comment]
@driskull Yeah...I think a close button and/or URL param can wait. For my purposes, I can just CSS hide the header in those full page apps for now.

**Comment by driskull** [comment]
What do you want the url param to be? `headless`, `no-header`, `header-disabled`?

**Comment by asangma** [comment]
`header-disabled` seems to match our pattern.

**Comment by driskull** [comment]
DONE

**Comment by asangma** [comment]
Yew! Thanks @driskull!

**Comment by driskull** [comment]
@julio8a can you approve?

**Comment by asangma** [comment]
@bstifle and @rmstinson were seeing a UI bug that @driskull and I are not seeing. I say we approve and murgle and see if the UI bug persists after murgling.

**Comment by asangma** [comment]
FYI. UI bug that @bstifle reported is not specific to this PR.

**Comment by bstifle** [comment]
Head is cool. but im not seeing the lines on dark theme:

image

image

**Comment by asangma** [comment]
@bstifle That issue is not specific to this PR. Suggest creating a new issue for what you're seeing.

**Comment by bstifle** [comment]
@caripizza @driskull related to this: https://github.com/Esri/calcite-components/pull/2772

still in the works?

**Comment by bstifle** [comment]
image
just noticed this too haha! Z-axis issue?

**Comment by asangma** [comment]
re: z-index issue.
I'll have a look at that.

**Comment by driskull** [comment]
Create separate issues for those. Lets not hold this up

**Comment by asangma** [comment]
@driskull Done.

**Comment by driskull** [comment]
The components that are going over the header need to have a `position:relative` on their host most likely. Their zIndex shouldn't be leaking out.

**Review by julio8a — APPROVED** [review]

**Pull request merged**
